.github/workflows/main.yml

```
@@ -6,79 +6,365 @@
       matrix:
-        os: [ ubuntu-latest, macos-latest, windows-latest ]
-        build_type: [ Release ]
+        # MacOS is disabled for now as TeDDy does not support it
+        os: [ubuntu-latest, windows-latest]
+        build_type: [Debug, Release]
+        compiler: [gcc, clang, msvc]
+        exclude:
+          # Exclude combinations that don't make sense
+          - os: ubuntu-latest
+            compiler: msvc
+          - os: macos-latest
```

actionlint (1.7.8)

value "macos-latest" in "exclude" does not match in matrix "os" combinations. possible values are "ubuntu-latest", "windows-latest"

(matrix)